Tests/Contour.Common.Tests/PatternsSpecs.cs

```diff
@@ -287,7 +287,7 @@ public void should_accept_message()
                     "Body");
 
                 Mock<IDelivery> deliveryMock = new Mock<IDelivery>();
-                BusProcessingContext.Current = new BusProcessingContext(deliveryMock.Object);
+                BusProcessingContext.Current = new BusProcessingContext(deliveryMock.Object, null);
```

Comment:
Bus processing context acts as an aggregate for delivery and bus, thus it should maintain it's consistency, e.x. delivery and bus context should both be null or not null.
In other way why should one provide the delivery and not the bus which is related to the delivery?

This is one more reason to push just the consuming context to the constructor and extract delivery and bus inside (assuming a potentially non-type safe cast to IDeliveryContext which should be fixed aside this task).